### DISCUSSION TITLE
Function declarations in std.prql

### DISCUSSION THREAD

**Discussion by broneill** [PRQL/prql#4288] [Q&A] [answered]
I'm looking at the definition of the `join` function in std.prql, and I'm confused:

```
let join = func
  with <relation>
  condition <bool>
  `noresolve.side`:inner
  tbl <relation>
  -> <relation> internal join
```

It declares the `side` parameter after the `condition` parameter, but doesn't the actual function accept the `side` parameter first?

**Comment by max-sixty** [comment, marked as answer]
`side` is a named parameter so can go anywhere! Check out the top item in https://prql-lang.org/book/reference/declarations/functions.html

**Reply by broneill** [reply]
Got it, thanks. Is there any reason why the `side` parameter is declared there, given that the convention is to provide the side first?

**Reply by max-sixty** [reply]
I don't think so, though I'm not 100% sure...
